rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2281,11 +2281,27 @@ extern "C"
 		return RMW_RET_ERROR;
 	}
 	//Get participant pointer from node
-        if(node->implementation_identifier != eprosima_fastrtps_identifier)
-        {
-            RMW_SET_ERROR_MSG("node handle not from this implementation");
-            return RMW_RET_ERROR;
-        }
+	if(node->implementation_identifier != eprosima_fastrtps_identifier)
+	{
+	    RMW_SET_ERROR_MSG("node handle not from this implementation");
+	    return RMW_RET_ERROR;
+	}
+	if (!topic_names_and_types) {
+	  RMW_SET_ERROR_MSG("topics handle is null");
+	  return RMW_RET_ERROR;
+	}
```

Comment:
Sorry about that: fixed in https://github.com/eProsima/ROS-RMW-Fast-RTPS-cpp/pull/59/commits/546ffa52cf8c99867e3d130a93545d1732d27fac